front/components/common/fields/Fields.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import styled from "styled-components";
+import { FieldErrors } from "react-hook-form";
+
+import DatePickerField, { DatePickerFieldProps } from "./DatePickerField";
+import InputField, { InputFieldProps } from "./InputField";
+import RadioField, { RadioFieldProps } from "./RadioField";
+import { THEME_COLOR1, FORM_ERROR_COLOR } from "../../../utils/theme/theme";
+
+type FieldsInterface = DatePickerFieldProps | InputFieldProps | RadioFieldProps;
+
+const Fields = (props: FieldsInterface) => {
+  const { type, name, control, error, prefix, required = false, ...rest } = props;
+
+  let ReturnComponent = InputField;
+
+  switch (type) {
+    case "date":
+      ReturnComponent = DatePickerField;
+      break;
+    case "radio":
+      ReturnComponent = RadioField;
+      break;
+  }
```

Comment:
`let ReturnComponent`를 일반 변수 선언이 아닌 `useState()`로 관리하는 건 어떻게 생각하세요? 사실 초기 렌더에만 결정되는 거라 상관은 없지만 통일성을 고려하면 사용하는 것도 나쁘지 않다고 생각합니다.

그리고 밑의 switch case문이 나중에 길어지는 걸 대비해서 컴포넌트 바깥에 두는건 어떻게 생각하시나요?
```suggestion
function getReturnComponent(type) {
  switch (type) {
    case "date":
      return DatePickerField;
    case "radio":
      return RadioField;
    default :
      return InputField;
  }
}

const Fields = (props: FieldsInterface) => {
  const { type, name, control, error, prefix, required = false, ...rest } = props;
  const [ReturnComponent, setReturnComponent] = useState(getReturnComponent(type));
```

Reply:
일반적으로 리액트 컴포넌트 자체를 state로 관리하지 않는걸로 알고 있는데요, 이유는 정확히 모르곘네요. 보통 컴포넌트 내에서 필요한 데이터를 관리하기 위해 state를 사용합니다. 
switch case 문을 컴포넌트 밖에 두는건 좋습니다. 반영할게요!

Reply:
넵 혹시 지금 바로 반영하시나요 ? 아니면 바로 머지해서 저도 작업 시작할게요.

Reply:
앗 반영완료했습니다. 머지할테니 작업시작하시면 됩니다!